src/library_pthread.js

```diff
@@ -41,7 +35,8 @@ var LibraryPThread = {
         PThread.allocateUnusedWorker();
       }
 #endif
-
+    },
+    initRuntime: function() {
       PThread.mainThreadBlock = {{{ makeStaticAlloc(C_STRUCTS.pthread.__size__) }}};
```

Comment:
After this change, the function `initMainThreadBlock` no longer initializes `PThread.mainThreadBlock` (but instead `initRuntime` does that). That naming mismatch is awkward, so maybe change name `initMainThreadBlock` to `loadPThreadPool` or something like that?

Reply:
@juj good point, thanks, I opened https://github.com/emscripten-core/emscripten/pull/12604 now.